src/Runner.Worker/ActionManager.cs

```diff
@@ -64,10 +77,30 @@ public async Task<PrepareResult> PrepareActionsAsync(IExecutionContext execution
                 PreStepTracker = new Dictionary<Guid, IActionRunner>()
             };
             var containerSetupSteps = new List<JobExtensionRunner>();
-            IOUtil.DeleteDirectory(HostContext.GetDirectory(WellKnownDirectory.Actions), executionContext.CancellationToken);
+            var depth = 0;
+            // We are running at the start of a job
+            if (rootStepId == default(Guid))
+            {
+                IOUtil.DeleteDirectory(HostContext.GetDirectory(WellKnownDirectory.Actions), executionContext.CancellationToken);
+            }
+            // We are running mid job due to a local composite action
+            else
+            {
+                if (!_cachedEmbeddedStepIds.ContainsKey(rootStepId))
```

Comment:
since we're adding `EmbeddedId` to execution context. Do we need this embedded ID dictionary now?

Reply:
we spoke offline, we do because we don't have access to the execution context at this point.